mirror-plugins

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env ruby
+
+require "rubygems"
+require "bundler/setup"
+require "dotenv"
+require "gitlab"
+
+Dotenv.load
+
+Gitlab.configure do |config|
+  config.endpoint = ENV["GITLAB_API_ENDPOINT"]
+  config.private_token = ENV["GITLAB_ACCOUNT_TOKEN"]
+end
+
+def github_repo_exists?(project)
+  system("git ls-remote -q #{construct_github_https_url(project)} >/dev/null 2>&1")
+end
+
+def gitlab_project_empty?(project)
+  gitlab_commits = Gitlab.commits(project.id, per_page: 1)
+  gitlab_commits.empty?
+end
+
+def repos_in_sync?(project)
+  github_latest_commit_hash = `gh api repos/#{ENV["GITHUB_ORG_NAME"]}/#{project.name}/commits/#{ENV["DEFAULT_BRANCH_NAME"]} -q '.sha'`
+  gitlab_commits = Gitlab.commits(project.id, per_page: 1)
+  gitlab_latest_commit_hash = gitlab_commits[0].id
+  github_latest_commit_hash.strip == gitlab_latest_commit_hash.strip
+end
+
+def construct_gitlab_https_url(project)
+  credentials = ENV["GITLAB_ACCOUNT_USERNAME"] + ":" + ENV["GITLAB_ACCOUNT_TOKEN"] + "@"
+  project.http_url_to_repo.insert(8, credentials)
+end
+
+def construct_github_https_url(project)
+  credentials = ENV["GITHUB_ACCOUNT_USERNAME"] + ":" + ENV["GITHUB_ACCOUNT_TOKEN"] + "@"
+  url = "https://github.com/" + ENV["GITHUB_ORG_NAME"] + "/" + project.name + ".git"
+  url.insert(8, credentials)
+end
+
+def mirror_gitlab_repo(project, create = false)
```

Comment:
I recommend avoiding boolean arguments that change the behaviour of the method. They increase complexity and the chance of error when calling it. In this case I'd prefer to define two methods. Depending on how the implementation works, that could be:

- a method to do just the sync and a method to do just the creation (eg `sync_mirror` and `create_mirror`); or
- a method to do just the sync and a method to do create and sync (eg `mirror` and `create_and_mirror`)

Reply:
I've split this out into two methods (`create_github_repo` and `update_github_mirror`) now.